README.adoc

```diff
@@ -1302,6 +1302,69 @@ end
 ====
 
 
+==== Encoding Options in XmlAdapter
+
+XmlAdapter supports the encoding in the following ways:
+
+. When encoding is not passed in to_xml:
+** Default encoding is UTF-8.
+
+. When encoding is explicitly passed nil:
+** Encoding will be nil, show the HexCode(Nokogiri) or ASCII-8bit(Ox).
+
+. When encoding is passed with some option:
+** Encoding option will be selected as passed.
+
+
+Syntax:
+
+[source,ruby]
+----
+Example.new(description: " ∑ is my ∏ moniker µ.").to_xml
+Example.new(description: " ∑ is my ∏ moniker µ.").to_xml(encoding: nil)
+Example.new(description: " ∑ is my ∏ moniker µ.").to_xml(encoding: "ASCII")
```

Comment:
Currently, we do not support an explicit parsing encoding option because the Adapter handles all encodings intelligently.

For Example:

```ruby
xml = "<name> John Doe </name>"

xml.force_encoding("UTF-16")
#=> "\x3C\x6E\x61\x6D\x65\x3E\x20\x4A\x6F\x68\x6E\x20\x44\x6F\x65\x20\x3C\x2F\x6E\x61\x6D\x65\x3E" 

Nokogiri::XML(xml.force_encoding("UTF-16"))
# => 
#  (Document:0x17250 {
#    name = "document",
#    children = [ #(Element:0x1737c { name = "name", children = [ #(Text " John Doe ")] })]
#  }) 

Ox.parse(xml.force_encoding("UTF-16"))
#=> #<Ox::Element:0x000000010f753ce8 @nodes=[" John Doe "], @value="name"> 
```

@ronaldtse Do we still need to add the encoding option?

Reply:
If the file uses an encoding like Latin or Shift-JIS, we won't be able to parse the file unless the encoding is given.